src/datahike/db.cljc

```diff
@@ -1035,6 +1035,25 @@
                         :temporal-aevt (di/empty-index index :aevt index-config)
                         :temporal-avet (di/empty-index index :avet index-config)}))))))
 
+(defn metrics [db]
+  {:pre [(instance? DB db)]}
+  (let [attr-entity-counts (into {} (map (fn [[a d]] [a (group-by :e d)]) (group-by :a (-seq (.-eavt db)))))
+        per-attr-counts
+        (into {} (map (fn [[a ent-ds]] [a (reduce + 0 (map count (vals ent-ds)))]) attr-entity-counts))
+        filter-idxed-attrs (fn [attr-kvs] (filter (fn [[k v]] (-> (.-rschema db) :db/index k)) attr-kvs))]
+    {:count (-count (.-eavt db))
+     :per-attr-counts per-attr-counts
+     :per-entity-counts (reduce (fn [oecs aecs]
+                                  (reduce-kv
+                                   (fn [ecs e c] (if (ecs e) (update ecs e #(+ % c)) (assoc ecs e c)))
+                                   oecs aecs))
+                                {}
+                                (map #(into {} (map (fn [[e d]] [e (count d)]) %)) (vals attr-entity-counts)))
```

Comment:
Hard to read: Which `map` has `(vals attr-entity-counts)` as parameter? The first or the second one? It's not obvious with just one look. Solution: threading macro or separate line for collection.